### PR TITLE
Rename Pokémon animation files; consistent (x, y) coords; document design flaws

### PR DESCRIPTION
**Pull request by roukaour**

### CODE REVIEW

**Comment by xCrystal** [comment]
Since you are messing with the bugs and glitches doc in this PR, here are some misc things before I forget about them:
- At some point it would be nice if they were arranged by category (e.g. battle engine, overworld engine, graphics, text, audio, code, etc.)
- This one is not covered: https://twitter.com/crystal_rby/status/874626362287562752
- `GetForestTreeFrame` should be moved to design flaws.

File: docs/design_flaws.md [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+	ld d, h
+	ld hl, VTiles2 tile $64
+	lb bc, BANK(Footprints), 2
+	call Request1bpp
+```
+
+
+## Pokédex entry banks are derived from their species IDs
+
+`PokedexDataPointerTable` in [data/pokemon/dex_entry_pointers.asm](/data/pokemon/dex_entry_pointers.asm) is a table of `dw`, not `dba`, yet there are four banks used for Pokédex entries. The correct bank is derived from the species ID at the beginning of each entry. (This is the only use that species ID has.)
```
Comment (yenatch) [inline docs/design_flaws.md]: should mention that this is the species id in base stats

Comment (yenatch) [inline docs/design_flaws.md]: should mention that this is the id in base stats

File: docs/design_flaws.md [hunk 2]
```diff
@@ -355,3 +442,70 @@ GetNumberedTMHM: ; d417
 	ld c, a
 	ret
 ```
+
+**Fix:**
+
+Move `ITEM_C3` and `ITEM_DC` above all the TMs in every table of item data.
```
Comment (yenatch) [inline docs/design_flaws.md]: These are catch rates used in RBY. You'll need to also modify the time capsule code to compensate.

**Pull request merged**